laboratory/src/org/labkey/laboratory/LaboratoryContainerListener.java

```diff
@@ -112,69 +108,11 @@ public void propertyChange(PropertyChangeEvent evt)
 
         if (evt.getPropertyName().equals(ContainerManager.Property.Name.name()))
         {
-            if (evt instanceof ContainerManager.ContainerPropertyChangeEvent)
-            {
-                ContainerManager.ContainerPropertyChangeEvent ce = (ContainerManager.ContainerPropertyChangeEvent) evt;
-                if (ce.container.isWorkbook())
-                {
-                    TableInfo ti = LaboratorySchema.getInstance().getTable(LaboratorySchema.TABLE_WORKBOOKS);
-                    TableSelector ts = new TableSelector(ti, new SimpleFilter(FieldKey.fromString(LaboratoryWorkbooksTable.WORKBOOK_COL), ce.container.getId()), null);
-                    if (ts.exists())
-                    {
-                        try
-                        {
-                            Integer rowId = Integer.parseInt(ce.container.getName());
-                            WorkbookModel w = ts.getObject(ce.container.getId(), WorkbookModel.class);
-                            w.setWorkbookId(rowId);
-                            Table.update(ce.user, ti, w, ce.container.getId());
-                        }
-                        catch (NumberFormatException e)
-                        {
-
-                        }
-                    }
-                }
-            }
+            updateWorkbookTableOnNameChange(evt);
         }
         else if (evt.getPropertyName().equals(ContainerManager.Property.Policy.name()))
```

Comment:
This should now be listening to the newly added Modules property instead of Policy, I think

Reply:
this exists in release20.7?

Reply:
https://github.com/LabKey/platform/blob/release20.7/api/src/org/labkey/api/data/ContainerManager.java

Reply:
It's in the snapshot branch. We merge that to the release branch itself when we ship an official version like 20.7.0.

https://github.com/LabKey/platform/commit/8a495ff64708d26a92e1ad1d481136a5f659387c